app/models/deploy_service.rb

```diff
@@ -13,6 +13,13 @@ def deploy!(stage, attributes)
     if deploy.persisted?
       send_sse_deploy_update('new', deploy)
 
+      if stage.cancel_queued_deploys?
+        stage.deploys.pending.prior_to(deploy).for_user(user).each do |queued_deploy|
+          JobExecution.dequeue(queued_deploy.job.id)
```

Comment:
only do the cancel if it is successfully dequeued ... the job you are looking at might already be running now

`if JobExecution.dequeue ... `